contracts/TimeManagerV8.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.13;
+
+import { SECONDS_PER_YEAR } from "./constants.sol";
+
+abstract contract TimeManagerV8 {
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable blocksOrSecondsPerYear;
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    bool public immutable isTimeBased;
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    function() view returns (uint256) private immutable _getCurrentSlot;
+
+    /// @notice Thrown when blocks per year is invalid
+    error InvalidBlocksPerYear();
+
+    /// @notice Thrown when time based but blocks per year is provided
```

Comment:
```suggestion
    /// @notice Thrown when time based but blocks per year is zero
```

Reply:
Error will be thrown when it is time-based and blocks per year are also provided

Reply:
Yeah, I mistyped.
We can say:
`/// @notice Thrown when time based but blocks per year is non zero`